apps/total-typescript/src/team/index.tsx

```diff
@@ -25,40 +25,55 @@ const InviteTeam: React.FC<React.PropsWithChildren<InviteTeamProps>> = ({
   setPersonalPurchase,
 }) => {
   const bulkCouponSchema = z
-    .object({maxUses: z.number(), usedCount: z.number()})
-    .transform(({maxUses, usedCount}) => {
+    .object({id: z.string(), maxUses: z.number(), usedCount: z.number()})
+    .nullable()
+    .transform((data, ctx) => {
+      if (data === null) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message:
+            'The purchase is either not a bulk purchase or was queried without its bulk coupon.',
+        })
+        return z.NEVER
+      }
```

Comment:
The `purchase.bulkCoupon` can potentially come in as `null`. This now accounts for that and surfaces a much better runtime error message if that is the case.